src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/AzureSecurityRulePlugin.java

```diff
@@ -0,0 +1,130 @@
+package cloud.fogbow.ras.core.plugins.interoperability.azure.securityrule;
+
+import cloud.fogbow.common.constants.AzureConstants;
+import cloud.fogbow.common.exceptions.FogbowException;
+import cloud.fogbow.common.exceptions.InvalidParameterException;
+import cloud.fogbow.common.exceptions.UnexpectedException;
+import cloud.fogbow.common.models.AzureUser;
+import cloud.fogbow.common.util.PropertiesUtil;
+import cloud.fogbow.ras.api.http.response.SecurityRuleInstance;
+import cloud.fogbow.ras.api.parameters.SecurityRule;
+import cloud.fogbow.ras.constants.Messages;
+import cloud.fogbow.ras.core.models.orders.Order;
+import cloud.fogbow.ras.core.plugins.interoperability.SecurityRulePlugin;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.securityrule.sdk.AzureNetworkSecurityGroupOperationSDK;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.securityrule.sdk.model.AzureUpdateNetworkSecurityGroupRef;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.securityrule.util.SecurityRuleIdContext;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.util.AzureGeneralUtil;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.util.AzureResourceIdBuilder;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+public class AzureSecurityRulePlugin implements SecurityRulePlugin<AzureUser> {
+
+    private static final Logger LOGGER = Logger.getLogger(AzureSecurityRulePlugin.class);
+
+    private final String defaultResourceGroupName;
+    private AzureNetworkSecurityGroupOperationSDK azureNetworkSecurityGroupOperationSDK;
+
+    public AzureSecurityRulePlugin(String confFilePath) {
+        Properties properties = PropertiesUtil.readProperties(confFilePath);
+        this.azureNetworkSecurityGroupOperationSDK = new AzureNetworkSecurityGroupOperationSDK();
+        this.defaultResourceGroupName = properties.getProperty(AzureConstants.DEFAULT_RESOURCE_GROUP_NAME_KEY);
+    }
+
+    @Override
+    public String requestSecurityRule(SecurityRule securityRule, Order majorOrder, AzureUser azureUser)
+            throws FogbowException {
+
+        LOGGER.info(Messages.Info.REQUESTING_INSTANCE_FROM_PROVIDER);
+        checkOrderType(majorOrder);
+
+        String networkSecurityGroupName = AzureGeneralUtil.defineInstanceId(majorOrder.getInstanceId());
```

Comment:
Use "defineResourceName" method instead of "defineInstanceId".
Another point is that you could explain why the majorOrder(PublicIp or Network) becomes a networkSecurityGroup. O could add a comentary.

Reply:
Done.